skafka/src/main/scala/com/evolutiongaming/skafka/producer/Producer.scala

```diff
@@ -146,26 +147,26 @@ object Producer {
 
         def block(record: ProducerRecordJ[Bytes, Bytes]) = {
 
-          def callbackOf(promise: Promise[RecordMetadataJ]) = new Callback {
-            def onCompletion(metadata: RecordMetadataJ, exception: Exception) = {
+          def callbackOf(deferred: Deferred[F, Either[Throwable, RecordMetadataJ]]): Callback = {
+            (metadata: RecordMetadataJ, exception: Exception) =>
               if (exception != null) {
-                promise.failure(exception)
+                Effect[F].toIO(deferred.complete(exception.asLeft)).unsafeRunSync()
```

Comment:
I'd improve this, so we have `Effect[F].toIO(...).unsafeRunSync()` in single place